recipe/meta.yaml

```diff
@@ -38,40 +44,36 @@ outputs:
 {% else %}
           - cudatoolkit >={{ cuda_compiler_version }},<{{ cuda_major + 1 }}
 {% endif %}
-{% if cdt_name == "cos7" %}       # [linux]
-          - __glibc >=2.17        # [linux]
-{% endif %}                       # [linux]
+          - __glibc >=2.17                           # [linux and cdt_name == "cos7"]
     requirements:
       build:      # [win]
         - m2-sed  # [win]
       host:
         # Needed to symlink libcuda into sysroot libs.
-        - {{ compiler("c") }}
-{% if cdt_name == "cos7" %}                    # [linux]
-        - sysroot_{{ target_platform }} 2.17   # [linux]
-        - __glibc >=2.17                       # [linux]
-{% endif %}                                    # [linux]
+        - {{ c_compiler }}_{{ cross_target_platform }} {{ c_compiler_version }}.*       # [linux]
+        - {{ c_compiler }}_{{ cross_target_platform }}                                  # [win]
```

Comment:
Should we version this for windows in the case that conda-forge wants to generally move to newer versions than what an older version of `nvcc` supports?

Reply:
I think that's unrelated to this PR and should be a separate discussion.